### PR TITLE
feat(atenlib):add ops(scalar tensor, constant_pad_nd)

### PR DESCRIPTION
**Pull request by xiaowuhu**

### CODE REVIEW

File: onnxscript/function_libs/torch_aten/ops/core.py [hunk 1]
```diff
@@ -1091,10 +1091,21 @@ def aten_conj_physical(self: TensorType) -> TensorType:
     raise NotImplementedError()
 
 
-def aten_constant_pad_nd(self: TensorType, pad: INT64, value: float = 0.0) -> TensorType:
+@torch_op("aten::constant_pad_nd", trace_only=True)
```
Comment (gramalingam) [inline onnxscript/function_libs/torch_aten/ops/core.py:1094]: Why does this need to be trace-only?

File: onnxscript/function_libs/torch_aten/ops/core.py [hunk 2]
```diff
@@ -1091,10 +1091,21 @@ def aten_conj_physical(self: TensorType) -> TensorType:
...
+def aten_constant_pad_nd(self: TTensor, pad: Sequence[INT64], value: float = 0.0) -> TTensor:
```
Comment (gramalingam) [inline onnxscript/function_libs/torch_aten/ops/core.py:1095]: What is the type of pad supposed to be? `Sequence[INT64]` looks unlikely. Is it really expected to be a list of tensors (containing a single element)? Or, it is expected to be a tensor? Or, a list of ints?

`INT64` indicates a tensor (of rank 0) of int64 value.

Comment (xiaowuhu) [reply onnxscript/function_libs/torch_aten/ops/core.py:1096]: should be INT64

File: onnxscript/function_libs/torch_aten/ops/core.py (hunk 1 above)
Comment (xiaowuhu) [reply onnxscript/function_libs/torch_aten/ops/core.py]: Thanks for the reminder. I made it with full onnx ops.

File: onnxscript/function_libs/torch_aten/ops/core.py [hunk 3]
```diff
@@ -1092,10 +1092,42 @@ def aten_conj_physical(self: TensorType) -> TensorType:
...
+    # assume zero-dimensions in the beginning
+    # rank = len(self.shape)  # rank must be scalar
+    # paddings = list(pad[:]) + [0] * (rank * 2 - len(pad))
+    # reverse order and collate first beginnings and then ends
+    # paddings = paddings[-2::-2] + paddings[-1::-2]
+
+    neg_1 = op.Constant(value_ints=[-1])
+
+    rank = op.Size(op.Shape(self))
+    zero_count = op.Sub(op.Mul(rank, op.Constant(value_int=2)), op.Size(pad))
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_aten/ops/core.py:1111]:
```suggestion
    zero_count = op.Sub(op.Mul(rank, 2), op.Size(pad))
```

for simplicity

File: onnxscript/function_libs/torch_aten/ops/core.py [hunk 4]
```diff
@@ -1092,10 +1092,42 @@ def aten_conj_physical(self: TensorType) -> TensorType:
...
+    zero_count = op.Reshape(zero_count, neg_1)
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_aten/ops/core.py:1112]:
```suggestion
```
is reshape needed?

File: onnxscript/function_libs/torch_aten/ops/core.py [hunk 5]
```diff
@@ -1092,10 +1092,42 @@ def aten_conj_physical(self: TensorType) -> TensorType:
...
+    odd_elements = op.Slice(torch_paddings, starts, ends, zero, steps)
+
+    starts = neg_1
+    ends = op.Sub(starts, size_d)
+    even_elements = op.Slice(torch_paddings, starts, ends, zero, steps)
+
+    onnx_padding = op.Concat(odd_elements, even_elements, axis=0)
+    result = op.Pad(self, onnx_padding, value)
+
+    return result
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_aten/ops/core.py:1130]:
```suggestion
    return op.Pad(self, onnx_padding, value)
```

for simplicity

**Review by justinchuby — COMMENTED** [review]
Looking great, thank you! Some minor comments

File: onnxscript/function_libs/torch_aten/ops/core.py (hunk 4 above)
Comment (justinchuby) [reply onnxscript/function_libs/torch_aten/ops/core.py:1112]: my bad

**Review by justinchuby — APPROVED** [review]

**Comment by justinchuby** [comment]
Will be great to get a second review from @fatcat-z

**Review by fatcat-z — APPROVED** [review]
Thank you!

**Pull request merged**